src/XMakeBuildEngine/Logging/BaseConsoleLogger.cs

```diff
@@ -370,7 +370,7 @@ internal bool IsVerbosityAtLeast(LoggerVerbosity checkVerbosity)
         /// <summary>
         /// Sets foreground color to color specified
         /// </summary>
-        internal static void SetColor(ConsoleColor c)
+        public static void SetColor(ConsoleColor c)
```

Comment:
Protected definitely wouldn't work since as you mention it's being called from the ConsoleLogger class. However, they're both in the same assembly so keeping it internal should just work. Let me know if I'm missing a scenario though. 

(Just to make sure I cloned your changes and switched to internal, all tests and the build passed with no issues)